resourcemanager/commonids/tenant.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package commonids
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+var _ resourceids.ResourceId = &TenantId{}
+
+// TenantId is a struct representing the Resource ID for a Tenant
+type TenantId struct {
+	TenantId string
+}
+
+// DefaultTenantId returns a default TenantId
+var DefaultTenantId = TenantId{
+	TenantId: "/",
+}
```

Comment:
pedantic request: but rather than defining the fields inline can we use `NewTenantID` here? It ensures that were we to add another field here in the future that we'd get a compiler error, so this'd be updated?

```suggestion
var DefaultTenantId = NewTenantID("/")
```

Reply:
makes sense 👍  I've updated that now